pkg/builder/snapbuilder/build.go

```diff
@@ -123,6 +123,13 @@ func (b *Builder) WithCapacity(capacity string) *Builder {
 	return b
 }
 
+// WithThinProvision sets if snapshot is thin snapshot based on the volume
+// provision type
+func (b *Builder) WithThinProvision(thinProvision string) *Builder {
+	b.snap.Object.Spec.ThinProvision = thinProvision
```

Comment:
We need to add a new field for thin provisioning of snapshot as done in the snapshot size change [here](https://github.com/openebs/lvm-localpv/pull/99/files?file-filters%5B%5D=.go#diff-eb7eaeadd94e0938d74bd901e6d51cc928abe1c5015f95c5a18caf6a19c588abR37-R42).  the volume spec stores the information about volume at the time of snapshotting. Ideally we should not modify it. 

Ref [this](https://github.com/openebs/lvm-localpv/issues/97#issuecomment-860843393) comment for more info.

cc: @pawanpraka1